lib/select.c

```diff
@@ -374,7 +376,6 @@ int Curl_poll(struct pollfd ufds[], unsigned int nfds, timediff_t timeout_ms)
   else
     pending_ms = 0;
   r = poll(ufds, nfds, pending_ms);
-
   if(r < 0)
     return -1;
```

Comment:
I basically just wanted to make our wrapping code do the same to select() and poll() return values. Both should already only return -1 on error according to POSIX and Linux man pages, so our own API is fulfilled by that already and we can drop the if block for both. I will update the PR for that, if you are find with this approach:
```
  if(r <= 0)
    return r;
```

Reply:
Yes, like select() poll() will also only return -1 on errors so that change looks fine.

Reply:
Now part of https://github.com/curl/curl/pull/5880